datafusion/spark/src/function/utils.rs

```diff
@@ -67,7 +69,12 @@ pub mod test {
                     let return_field = return_field.unwrap();
                     assert_eq!(return_field.data_type(), &$EXPECTED_DATA_TYPE);
 
-                    let result = func.invoke_with_args(datafusion_expr::ScalarFunctionArgs{args: $ARGS, number_rows: cardinality, return_field: &return_field, arg_fields: arg_fields_owned.iter().collect()});
+                    let result = func.invoke_with_args(datafusion_expr::ScalarFunctionArgs{
+                        args: $ARGS,
+                        number_rows: cardinality,
+                        return_field: &return_field,
+                        arg_fields: arg_fields.clone(),
```

Comment:
in particular the change is to avoid calling `collect` twice and instead create the vec once and reuse it